src/test/ui/token/issue-15980.stderr

```diff
@@ -0,0 +1,26 @@
+error: expected identifier, found keyword `return`
+  --> $DIR/issue-15980.rs:20:13
+   |
+18 |         Err(ref e) if e.kind == io::EndOfFile {
+   |                                 ------------- while parsing this struct
+19 |             //~^ NOTE while parsing this struct
+20 |             return
+   |             ^^^^^^ expected identifier, found keyword
```

Comment:
This was the only line originally being emitted for this input. Now it correctly complains about missing `=>` below, as well as continue to type checking.